library/src/main/java/com/novoda/downloadmanager/lib/DownloadThread.java

```diff
@@ -343,6 +346,7 @@ private void runInternal() {
      * handle the response, and transfer the data to the destination file.
      */
     private void executeDownload(State state) throws StopRequestException {
+        checkPausedOrCanceled(originalDownloadInfo);
```

Comment:
@zegnus Why is this checked here as well ? This case should never happen right ?